Ingress/controllers/haproxy/README.md

```diff
@@ -0,0 +1,310 @@
+# Bare Metal Service Load Balancers
+
+AKA "how to set up a bank of haproxy for platforms that don't have load balancers".
+
+## Disclaimer:
+- This is a **work in progress**.
+- A better way to achieve this will probably emerge once discussions on (#260, #561) converge.
+- Backends are pluggable, but [Haproxy](https://cbonte.github.io/haproxy-dconv/configuration-1.5.html) is the only loadbalancer with a working implementation.
+- I have never deployed haproxy to production, so contributions are welcome (see [wishlist](#wishlist) for ideas).
+- For fault tolerant load balancing of ingress traffic, you need:
+  1. Multiple hosts running load balancers
+  2. Multiple A records for each hostname in a DNS service.
+
+This module will not help with the latter
+
```

Comment:
This readme needs to be re-written. It should describe both modes of the Ingress controller (services and ingress) and give some example ingressese that one can use to verify it works. I'm fine with punting this to a follow up pr, just add:
```
## TODO 
This controller has gone through many changes and now supports both [Ingress](https://github.com/kubernetes/kubernetes/blob/release-1.1/docs/user-guide/ingress.md) and [Services](https://github.com/kubernetes/kubernetes/blob/release-1.1/docs/user-guide/services.md). 
The documentation below is valid when it is started with --use-ingress=false, this is the default, but this will change very soon.
```
at the top.